pennylane/templates/base/__init__.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2018-2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Base templates serve as building blocks for other templates. They define a fundamental structure of how to arrange
```

Comment:
Continuing the thought from my previous comment, the base templates aren't templates themselves --- so perhaps a better name makes sense.

Reply:
I disagree - when used with ``unitary=RX`` or so, they are just ordinary templates.

Reply:
Yes, I see your point now. I guess when I think of templates, I think of a function where the structure/gates are already pre-determined, and modified by high-level parameters. With these, for some reason, I interpret them as accepting a template, and then outputting a different template...

Reply:
In this case I feel like @josh146